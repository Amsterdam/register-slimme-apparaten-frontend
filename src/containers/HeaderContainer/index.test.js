import React from 'react';
import { shallow } from 'enzyme';

import { isAuthenticated } from 'shared/services/auth/auth';
import { HeaderContainer } from './index';
// import Event from ;

jest.mock('shared/services/auth/auth');

describe('<HeaderContainer />', () => {
  let props;
  const event = {
    persist: jest.fn(),
    preventDefault: jest.fn(),
    stopPropagation: jest.fn(),
  };

  beforeEach(() => {
    props = {
      userName: 'user',
      onLogin: jest.fn(),
      onLogout: jest.fn()
    };
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly when authenticated', () => {
    isAuthenticated.mockImplementation((() => true));

    const renderedComponent = shallow(
      <HeaderContainer {...props} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  describe('onLoginLogoutButtonClick', () => {
    it('should login when not authenticated', () => {
      isAuthenticated.mockImplementation((() => false));

      const renderedComponent = shallow(
        <HeaderContainer {...props} />
      );

      const domain = 'the-login-domain';
      expect(renderedComponent.instance().onLoginLogoutButtonClick(event, domain));
      expect(props.onLogin).toHaveBeenCalledWith(domain);
    });

    it('should logout when authenticated', () => {
      isAuthenticated.mockImplementation((() => true));

      const renderedComponent = shallow(
        <HeaderContainer {...props} />
      );
      expect(renderedComponent.instance().onLoginLogoutButtonClick(event));
      expect(props.onLogout).toHaveBeenCalled();
    });
  });
});
