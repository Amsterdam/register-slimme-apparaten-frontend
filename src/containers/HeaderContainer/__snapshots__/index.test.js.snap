// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderContainer /> should render correctly when authenticated 1`] = `
<HeaderContainer__HeaderWrapper>
  <Header
    fullWidth={true}
    homeLink="/"
    logo={[Function]}
    navigation={
      <React.Fragment>
        <ForwardRef(MenuItem)>
          <MenuButton
            forwardedAs="a"
            href="/"
          >
            Kaart
          </MenuButton>
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)>
          <MenuButton
            forwardedAs="a"
            href="/categories"
          >
            Type apparaten
          </MenuButton>
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)>
          <MenuButton
            forwardedAs="a"
            href="/about/faq"
          >
            Veelgevraagd
          </MenuButton>
        </ForwardRef(MenuItem)>
        <ForwardRef(MenuItem)>
          <MenuButton
            forwardedAs="a"
            href="/about"
          >
            Over dit register
          </MenuButton>
        </ForwardRef(MenuItem)>
      </React.Fragment>
    }
    tall={false}
    title="Register slimme apparaten"
  />
</HeaderContainer__HeaderWrapper>
`;
